common-pg/src/main/scala/com/socrata/pg/store/PGSecondaryDatasetMapReader.scala

```diff
@@ -1,53 +1,167 @@
+       dataset_map.locale_name         as dataset_map_locale_name,
+       dataset_map.obfuscation_key     as dataset_map_obfuscation_key,
+       dataset_map.resource_name       as dataset_map_resource_name,
+       copy_map.system_id as copy_map_system_id,
+       copy_map.copy_number as copy_map_copy_number,
+       copy_map.lifecycle_stage :: TEXT as copy_map_lifecycle_stage,
+       copy_map.data_version as copy_map_data_version,
+       copy_map.data_shape_version as copy_map_data_shape_version,
+       copy_map.last_modified as copy_map_last_modified,
+       copy_map_table_modifiers.table_modifier as copy_map_table_modifiers_table_modifier
+from rollup_relationship_map
+         join rollup_map on rollup_relationship_map.rollup_system_id = rollup_map.system_id
+         join copy_map on rollup_map.copy_system_id = copy_map.system_id
+         left join copy_map_table_modifiers ON copy_map.system_id = copy_map_table_modifiers.copy_system_id
+         join dataset_map on copy_map.dataset_system_id = dataset_map.system_id
+where referenced_copy_system_id = ?"""
+
+  def getRollupsRelatedToCopy(copyInfo: CopyInfo): Set[LocalRollupInfo] = {
+    var out = Set[LocalRollupInfo]()
+    using(conn.prepareStatement(getRollupsRelatedToCopyQuery)) { stmt =>
+      stmt.setLong(1, copyInfo.systemId.underlying)
+      using(t("getRollupsRelatedToCopy", "copy_id" -> copyInfo.systemId)(stmt.executeQuery())) { rs =>
+        while (rs.next()) {
+          out += new LocalRollupInfo(
+            CopyInfo(
+              DatasetInfo(new DatasetId(rs.getLong("dataset_map_system_id")), rs.getLong("dataset_map_next_counter_value"), rs.getLong("dataset_map_latest_data_version"), rs.getString("dataset_map_locale_name"), rs.getBytes("dataset_map_obfuscation_key"), Option(rs.getString("dataset_map_resource_name"))),
+              new CopyId(rs.getLong("copy_map_system_id")),
+              rs.getLong("copy_map_copy_number"),
+              LifecycleStage.valueOf(rs.getString("copy_map_lifecycle_stage")),
+              rs.getLong("copy_map_data_version"),
+              rs.getLong("copy_map_data_shape_version"),
+              new DateTime(rs.getTimestamp("copy_map_last_modified").getTime),
+              Some(rs.getLong("copy_map_table_modifiers_table_modifier"))
+            ),
+            new RollupName(rs.getString("rollup_map_name")),
+            rs.getString("rollup_map_soql"),
+            rs.getString("rollup_map_table_name"),
+            RollupId(rs.getLong("rollup_map_system_id")))
+        }
+      }
+    }
+    out
+  }
+
+  def nextSequenceValue = "select nextval(?)"
+  def getNextSequence[T](sequenceName: String): Option[T] = {
+    using(conn.prepareStatement(nextSequenceValue)) { stmt =>
+      stmt.setString(1, sequenceName)
+      using(t("getNextSequence", "sequence" -> sequenceName)(stmt.executeQuery())) { rs =>
+        if (rs.next()) Some(rs.getObject(1)).asInstanceOf[Option[T]] else None
```

Comment:
`        if (rs.next()) Some(rs.getObject(1).asInstanceOf[T]) else None` is more direct. it's the object we're casting, not the Option.

Reply:
Has to be optional because you cannot guarantee a primitive will exist, or that the sequence exists.

Reply:
I think you can. You'll need to more the asInstanceOf into the Some.
Try compiling with the exact line I gave above.

Reply:
oh wait I misread this. Okay I agree